internal/handlers/handler_verify.go

```diff
@@ -232,7 +233,15 @@ func verifySessionCookie(ctx *middlewares.AutheliaCtx, targetURL *url.URL, userS
 	return userSession.Username, userSession.DisplayName, userSession.Groups, userSession.Emails, userSession.AuthenticationLevel, nil
 }
 
-func handleUnauthorized(ctx *middlewares.AutheliaCtx, targetURL fmt.Stringer, username string) {
+func handleUnauthorized(ctx *middlewares.AutheliaCtx, targetURL fmt.Stringer, isBasicAuth bool, username string) {
+	if isBasicAuth {
+		ctx.Logger.Infof("Access to %s is not authorized to user %s, sending 401 response with basic auth header", targetURL.String(), username)
+		ctx.ReplyUnauthorized()
+		ctx.Response.Header.Add("WWW-Authenticate", "Basic realm=\"Authentication required\"")
```

Comment:
Nginx forwards this header if you don't remap the 401 to something else ([docs](https://nginx.org/en/docs/http/ngx_http_auth_request_module.html)). I haven't checked other servers.

Also this is what solves my personal need, though I can always add the header from the reverse proxy, IMO it feels cleaner for authelia to inform the client we're expecting basic auth.

Reply:
I'm not averse to it staying. Though from what I understand that header is what should trigger the basic auth login prompt.

I never actually saw that occur, and I am using nginx.

Have you tested this and been able to replicate the behaviour?

Reply:
AFAIK authelia wasn't sending the header previously so yeah you shouldn't have seen it before.

Yes, I've been using this on my VPS since I created the PR. I just had to tweak my config to return a 401 instead of the 302 to the login page when using `/api/verify?auth=basic` to get the prompt, but that's expected. I'll include the details in the docs when I can get my workstation back up (hopefully this week-end).
The auto publishing of the image is awesome from a contributor perspective BTW!

Reply:
> AFAIK authelia wasn't sending the header previously so yeah you shouldn't have seen it before.

Yeah I didn't mean beforehand, I meant when I was testing it with your changes.

> The auto publishing of the image is awesome from a contributor perspective BTW!

Thanks for the feedback, we definitely would like to make the contributors experience seamless so feedback, good or bad as long as it's constructive is helpful :+1: .

Reply:
IIRC I noticed an error with that code only after testing on my server with the docker image, so you might've tested before it was working.